### PR TITLE
Validation function for ScoreCardDefinition

### PR DESCRIPTION
**Pull request by shrutiparabgoogle**

### CODE REVIEW

File: cmd/registry/scoring/parser_test.go [hunk 1]
```diff
@@ -1389,3 +1408,138 @@ func TestValidateBooleanThresholds(t *testing.T) {
...
 }
+
+func TestValidateScoreCardDefinition(t *testing.T) {
+	tests := []struct {
+		desc                string
+		parent              string
+		scoreCardDefinition *rpc.ScoreCardDefinition
+		wantNumErr          int
+	}{
+		// No errors
```
Comment (seaneganx) [inline cmd/registry/scoring/parser_test.go:1419]: Can these comments be incorporated into the test descriptions?

File: cmd/registry/scoring/parser_test.go [hunk 2]
```diff
@@ -1389,3 +1408,138 @@ func TestValidateBooleanThresholds(t *testing.T) {
...
+			wantNumErr: 4,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			ctx := context.Background()
+			gotErrs := ValidateScoreCardDefinition(ctx, test.parent, test.scoreCardDefinition)
+			if len(gotErrs) != test.wantNumErr {
+				t.Errorf("ValidateScoreCardDefinition() returned unexpected no. of errors: want %d, got %s", test.wantNumErr, gotErrs)
```
Comment (seaneganx) [inline cmd/registry/scoring/parser_test.go:1540]: This error message could include the function arguments to be more informative.

**Review by seaneganx — APPROVED** [review]

File: cmd/registry/scoring/parser_test.go (hunk 2 above)
Comment (shrutiparabgoogle) [reply cmd/registry/scoring/parser_test.go]: Done.

File: cmd/registry/scoring/parser_test.go (hunk 1 above)
Comment (shrutiparabgoogle) [reply cmd/registry/scoring/parser_test.go:1419]: These comments are just to organize the testcases. If we add these in test descriptions, it makes them too long, and I think the current test descriptions are quite informative.

**Review by timburks — APPROVED** [review]

**Pull request merged**